File: Taller_POO(Diagradmas)/Código_Ejercicio_Python.py
class Persona:
    def __init__(self,edad,nombre,apellido): #__init__ contructor
        self.edad=edad
        self.nombre=nombre
        self.apellido=apellido
    def info(self):
        return f"{self.edad},{self.nombre},{self.apellido}" #return para mostrar la información de la clase anterior

class Empleado(Persona):
    def __init__(self,edad,nombre,apellido,horario,cargo,salario):
        super().__init__(edad, nombre, apellido) #Llama al constructor de la clase Persona
        self.horario=horario
        self.cargo=cargo
        self.salario=salario

    def Realizar_venta(self):
        return f"El empleado {self.nombre} ha realizado una venta"
    def Atender_clientes(self):
        return f"El empleado {self.nombre} está atendiento clientes"
    def Info(self):
        return f"{self.edad},{self.nombre},{self.apellido},{self.horario},{self.cargo},{self.salario}"

class Administrador(Persona):
    def __init__(self,edad,nombre,apellido):
        super().__init__(edad, nombre, apellido)

    def Organizar_Roles(self):
        return f"El administrador{self.nombre} a organizado los roles"
    def Atender_clientes(self):
        return f"El empleado {self.nombre} está atendiento clientes"
    def Info(self):
        return f"{self.edad},{self.nombre},{self.apellido}"


class Cliente(Persona):
    def __init__(self,edad,nombre,apellido,dirección,teléfono,cedula,codigo_cliente):
        super().__init__(edad, nombre, apellido)
        self.dirección=dirección
        self.teléfono=teléfono
        self.cedula=cedula
        self.codigo_cliente=codigo_cliente

    def Realizar_compra(self):
        return f"El cliente{self.nombre} ha realizado una compra"
    def Info(self):
        return f"{self.edad},{self.nombre},{self.apellido},{self.dirección},{self.teléfono},{self.cedula},{self.codigo_cliente}"

class Venta:
    def __init__(self,Hora_venta,Día_venta,Mes_venta,Año_venta,Cantidad_productos_vendidos):
        self.Hora_venta=Hora_venta
        self.Día_venta=Día_venta
        self.Mes_venta=Mes_venta
        self.Año_venta=Año_venta
        self.Cantidad_productos_vendidos=Cantidad_productos_vendidos

    def Calcular_Total(self):
        return f"La venta ha calculado el total"
    def Generar_Factura(self):
        return f"La venta ha generado la factura"
    def Info(self):
        return f"{self.Hora_venta},{self.Día_venta},{self.Mes_venta},{self.Año_venta},{self.Cantidad_productos_vendidos}"   
class Inventario:
    def __init__(self,Lista_Prenda_Ropa,Stock):
        self.lista_Prenda_Ropa=Lista_Prenda_Ropa
        self.Stock=Stock
    def Generar_Reporte(self):
            return f"El inventario ha generado un reporte"
    def Actualizar_Stock(self):
            return f"El inventario ha actualizado el stock"
    def Info(self):
            return f"{self.lista_Prenda_Ropa},{self.Stock}"


#Ingresa información para probar las clases y sus métodos
if __name__ == "__main__": #Main para evitar una ejecución no deseada al tener las clases en la misma ruta
    empleado1 = Empleado("25", "Ana", "García", "Mañana", "Vendedora", "1500000")
    print(empleado1.Info())
    print(empleado1.Realizar_venta())
    print(empleado1.Atender_clientes())

    admin1 = Administrador("40", "Luis", "Martínez")
    print(admin1.Info())
    print(admin1.Organizar_Roles())
    print(admin1.Atender_clientes())
    
    cliente1 = Cliente("20", "Alejandro", "Zuñiga", "Cra 5 # 10-20", "3001234567", "123456798", "C001")
    print(cliente1.Realizar_compra())
    print(cliente1.Info())
    
    Venta1 = Venta("3:00", "20", "Enero", "2025", "5")
    print(Venta1.Calcular_Total())
    print(Venta1.Generar_Factura())
    print(Venta1.Info())
    
    Inventario1= Inventario(["Camisas, Pantalones, Chaquetas"], "100")
    print(Inventario1.Generar_Reporte())
    print(Inventario1.Actualizar_Stock())
    print(Inventario1.Info())

File: Taller_POO/Código_Ejercicio_Python.java
/*El código no ejecuta, solo es prueba de mi intento al realizar el código en lenguaje Java */
class Persona {
    protected String edad;  /*protegido:Acceso dentro de la clase y sus subclases*/
    protected String nombre; 
    protected String apellido;

    public Persona(String edad, String nombre, String apellido) { /*Public:Acceso al programa */
        this.edad = edad;
        this.nombre = nombre;
        this.apellido = apellido;
    }

    public String info() {
        return edad + "," + nombre + "," + apellido; /*Return devuelve un valor y termina la ejecución de un método*/
    }
}

class Empleado extends Persona {  /*Extends define que una clase es heredada de otra*/ 
    private String horario; /*Premite acceso solo dentro de la clase*/
    private String cargo;
    private String salario;

    public Empleado(String edad, String nombre, String apellido, String horario, String cargo, String salario) {
        super(edad, nombre, apellido);
        this.horario = horario;
        this.cargo = cargo;
        this.salario = salario;
    }

    public String realizarVenta() {
        return "El empleado " + nombre + " ha realizado una venta";
    }

    public String atenderClientes() {
        return "El empleado " + nombre + " está atendiendo clientes";
    }

    public String info() {
        return edad + "," + nombre + "," + apellido + "," + horario + "," + cargo + "," + salario;
    }
}

class Administrador extends Persona {
    public Administrador(String edad, String nombre, String apellido) {
        super(edad, nombre, apellido); /*Super:Hace referencia a el constructor*/
    }

    public String organizarRoles() {
        return "El administrador " + nombre + " ha organizado los roles";
    }

    public String atenderClientes() {
        return "El empleado " + nombre + " está atendiendo clientes";
    }

    public String info() {
        return edad + "," + nombre + "," + apellido;
    }
}

class Cliente extends Persona {
    private String direccion;
    private String telefono;
    private String cedula;
    private String codigoCliente;

    public Cliente(String edad, String nombre, String apellido, String direccion, String telefono, String cedula, String codigoCliente) {
        super(edad, nombre, apellido);
        this.direccion = direccion;
        this.telefono = telefono;
        this.cedula = cedula;
        this.codigoCliente = codigoCliente;
    }

    public String realizarCompra() {
        return "El cliente " + nombre + " ha realizado una compra";
    }

    public String info() {
        return edad + "," + nombre + "," + apellido + "," + direccion + "," + telefono + "," + cedula + "," + codigoCliente;
    }
}

class Venta {
    private String horaVenta;
    private String diaVenta;
    private String mesVenta;
    private String añoVenta;
    private String cantidadProductosVendidos;

    public Venta(String horaVenta, String diaVenta, String mesVenta, String añoVenta, String cantidadProductosVendidos) {
        this.horaVenta = horaVenta;
        this.diaVenta = diaVenta;
        this.mesVenta = mesVenta;
        this.añoVenta = añoVenta;
        this.cantidadProductosVendidos = cantidadProductosVendidos;
    }

    public String calcularTotal() {
        return "La venta ha calculado el total";
    }

    public String generarFactura() {
        return "La venta ha generado la factura";
    }

    public String info() {
        return horaVenta + "," + diaVenta + "," + mesVenta + "," + añoVenta + "," + cantidadProductosVendidos;
    }
}

class Inventario {
    private ArrayList<String> listaPrendaRopa; /*arraylist se utiliza para almacenar información en una lista*/
    private String stock;

    public Inventario(ArrayList<String> listaPrendaRopa, String stock) {
        this.listaPrendaRopa = listaPrendaRopa;
        this.stock = stock;
    }

    public String generarReporte() {
        return "El inventario ha generado un reporte";
    }

    public String actualizarStock() {
        return "El inventario ha actualizado el stock";
    }

    public String info() {
        return listaPrendaRopa.toString() + "," + stock;
    }
}

public class Código_Ejercicio_Python {
    public static void main(String[] args) { /*static:El método pertenece a la clase,no a los objetos, void no retorna ningun valor, main para ejecutar código*/
        Empleado empleado1 = new Empleado("25", "Ana", "García", "Mañana", "Vendedora", "1500000");
        System.out.println(empleado1.info());
        System.out.println(empleado1.realizarVenta());
        System.out.println(empleado1.atenderClientes());

        Administrador admin1 = new Administrador("40", "Luis", "Martínez");
        System.out.println(admin1.info()); /* system.out.println se utiliza para mostrar lso resultados y la información de objetos*/
        System.out.println(admin1.organizarRoles());
        System.out.println(admin1.atenderClientes());

        Cliente cliente1 = new Cliente("20", "Alejandro", "Zuñiga", "Cra 5 # 10-20", "3001234567", "123456798", "C001");
        System.out.println(cliente1.realizarCompra());
        System.out.println(cliente1.info());

        Venta venta1 = new Venta("3:00", "20", "Enero", "2025", "5");
        System.out.println(venta1.calcularTotal());
        System.out.println(venta1.generarFactura());
        System.out.println(venta1.info());

        ArrayList<String> prendas = new ArrayList<>();
        prendas.add("Camisas");
        prendas.add("Pantalones");
        prendas.add("Chaquetas");
        Inventario inventario1 = new Inventario(prendas, "100");
        System.out.println(inventario1.generarReporte());
        System.out.println(inventario1.actualizarStock());
        System.out.println(inventario1.info());
    }
}